api/bucketsd/service.go

```diff
@@ -1950,6 +1958,7 @@ func (s *Service) PushPaths(server pb.APIService_PushPathsServer) error {
 	if buckRoot != "" && buckRoot != buck.Path {
 		return status.Error(codes.FailedPrecondition, buckets.ErrNonFastForward.Error())
 	}
+	verifyBuck := buck.Copy()
```

Comment:
Copy the bucket instance for use in the queue. When a file is added to the queue, we just need to check if the requestor is allowed to make the change. We can re-add the changes serially when the file is done being added to IPFS, i.e., we get a result in `addedCh`.

Reply:
Cool, so making a copy to avoid concurrent access.

Maybe it would be a good idea to later rename this variable to "readOnlyBuck" or something that can hint a future development that it shouldn't use this *copy* for anything but reading. Now that in the same scope are two buckets (one copied and one the real one to save), it might be error-prone to do *new* changes in the wrong one.